spaceship.zsh

```diff
@@ -405,6 +416,73 @@ spaceship_time() {
     "$SPACESHIP_TIME_SUFFIX"
 }
 
+# BATTERY
+spaceship_battery() {
+  [[ $SPACESHIP_BATTERY_SHOW == false ]] && return
+
+  local battery_data battery_percent battery_status battery_color
+
+  if _exists "pmset"; then
+    battery_data=$(pmset -g batt)
+
+    # Return if no internal battery
+    [[ -z $(echo $battery_data | grep "InternalBattery") ]] && return
+
+    battery_percent="$( echo $battery_data | grep -oE '[0-9]{1,3}%' )"
+    battery_status="$( echo $battery_data | awk -F '; *' 'NR==2 { print $2 }' )"
+  elif _exists "upower"; then
+    local battery=$(command upower -e | grep battery | head -1)
+
+    # Return if no battery
+    [[ -z $battery ]] && return
+
+    battery_data=$(upower -i $battery)
+    battery_percent="$( echo $battery_data | grep percentage | awk '{print $2}' )"
+    battery_status="$( echo $battery_data | grep state | awk '{print $2}' )"
+  elif _exists "acpi"; then
+    battery_data=$(acpi -b)
+
+    # Return if no battery
+    [[ -z $battery_data ]] && return
+    battery_percent="$( echo $battery_data | awk '{print $4}' )"
+    battery_status="$( echo $battery_data | awk '{print tolower($3)}' )"
+  fi
+
+  # Remove trailing % and symbols for comparison
+  battery_percent="$(echo $battery_percent | tr -d '%[,;]')"
+
+  # Change color based on battery percentage
+  if [[ $battery_percent == 100 || $battery_status =~ "(charged|full)" ]]; then
+    battery_color="green"
+  elif [[ $battery_percent < $SPACESHIP_BATTERY_THRESHOLD ]]; then
+    battery_color="red"
+  else
+    battery_color="yellow"
+  fi
+
+  # Battery indicator based on current status of battery
+  if [[ $battery_status == "charging" ]];then
+    battery_symbol="${SPACESHIP_BATTERY_CHARGING_SYMBOL}"
+  elif [[ $battery_status =~ "^[dD]ischarg.*" ]]; then
+    battery_symbol="${SPACESHIP_BATTERY_DISCHARGING_SYMBOL}"
+  else
+    battery_symbol="${SPACESHIP_BATTERY_FULL_SYMBOL}"
+  fi
+
+  # Show section only if either of follow is true
```

Comment:
Move these comments to the top of the section to make it looks like other sections.